Packs/AWS-SNS/Integrations/AWSSNS/AWSSNS.py

```diff
@@ -0,0 +1,407 @@
+            data.append({'SubscriptionArn': subscription['SubscriptionArn']})
+
+        ec = {'AWS.SNS.Subscriptions': data}
+        return create_entry('AWS SNS Subscriptions', data, ec)
+
+    except Exception as e:
+        return raise_error(e)
+
+
+def send_message(args, client):
+    try:
+        data = []
+        kwargs = {
+            'Message': args.get('message')
+        }
+
+        if args.get('topicArn') is not None:
+            kwargs.update({'TopicArn': args.get('topicArn')})
+        if args.get('targetArn') is not None:
+            kwargs.update({'TargetArn': args.get('targetArn')})
+        if args.get('phoneNumber') is not None:
+            kwargs.update({'PhoneNumber': args.get('phoneNumber')})
+        if args.get('subject') is not None:
+            kwargs.update({'Subject': args.get('subject')})
+        if args.get('messageStructure') is not None:
+            kwargs.update({'MessageStructure': args.get('messageStructure')})
+        if args.get('messageDeduplicationId') is not None:
+            kwargs.update({'MessageDeduplicationId': args.get('messageDeduplicationId')})
+        if args.get('messageGroupId') is not None:
+            kwargs.update({'MessageGroupId': args.get('messageGroupId')})
+
+        response = client.publish(**kwargs)
+        data.append({'MessageId': response['MessageId']})
+        ec = {'AWS.SNS.SentMessages': data}
+        return create_entry('AWS SNS sent messages', data, ec)
+
+    except Exception as e:
+        return raise_error(e)
+
+
+def create_topic(args, client):
+    try:
+        attributes = {}
+        kwargs = {'Name': args.get('topicName')}
+        if args.get('deliveryPolicy') is not None:
+            attributes.update({'DeliveryPolicy': args.get('deliveryPolicy')})
+        if args.get('displayName') is not None:
+            attributes.update({'DisplayName': args.get('displayName')})
+        if args.get('fifoTopic') is not None:
+            attributes.update({'FifoTopic': bool(args.get('fifoTopic'))})
+        if args.get('policy') is not None:
+            attributes.update({'policy': args.get('Policy')})
+        if args.get('kmsMasterKeyId') is not None:
+            attributes.update({'KmsMasterKeyId': args.get('kmsMasterKeyId')})
+        if args.get('contentBasedDeduplication') is not None:
+            attributes.update({'ContentBasedDeduplication': args.get('contentBasedDeduplication')})
+        if attributes:
+            kwargs.update({'Attributes': attributes})
+
+        response = client.create_topic(**kwargs)
+        data = ({'ARN': response['TopicArn']})
+        ec = {'AWS.SNS.Topic': data}
+        return create_entry('AWS SNS Topic', data, ec)
+
+    except Exception as e:
+        return raise_error(e)
+
+
+def delete_topic(args, client):
+    try:
+        response = client.delete_topic(TopicArn=args.get('topicArn'))
+        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+            return 'The Topic has been deleted'
+
```

Comment:
What will be here if status != 200?

Reply:
@ilappe In the != 200 case, that will fall into exception case.